### DISCUSSION TITLE
How to query many-to-many relation with table referenced multiple times?

### DISCUSSION THREAD

**Discussion by Zebnastien** [supabase/supabase#21352] [Questions] [answered]
Let's consider these tables:

```js
Tasks {
   id : uuid
   author_id : uuid
}

Users {
   id : uuid
   name : text
}

TasksAccountables {
   user_id : uuid
   task_id : uuid
}
```

Since TaskAccount has a composite key, I should be able to query my tasks this way:

```js
await supabase.from("Tasks").select("*, Users(*)")
```

But because **Tasks.author_id** also references **Users** table, this query returns **null**.
What would be the correct way to do this query to get:

```js
Tasks {
   id : uuid
   author : Users
   accountables : Users[ ]
}
```

**Comment by GaryAustin1** [comment, marked as answer]
This talks about dealing with multiple keys.  https://postgrest.org/en/v12/references/api/resource_embedding.html#foreign-key-joins-on-multiple-foreign-key-relationships

But it does not specifically mention many-to-many as one of them.

You should also look at the error status returned as many times it suggests the best way to solve an error.

If not I would try using the hints to tell it to try the column user_id in TaskAccountables as a hint.

**Reply by Zebnastien** [reply]
You were totally right!

I did not figure I had a message in the response (response was not marked as error).

![image](https://github.com/supabase/supabase/assets/64729974/bddc85a5-d904-4ce6-8848-103d0f4d589e)

So the correct way to do the above mentioned query simply is:
```js
await supabase.from("Tasks").select("*, Users!TasksAccountables(*)")
```

And also return types are correct. Marvellous!
Thank you very much :)

**Reply by danieltistoj** [reply]
Muchas gracias
